main.go

```diff
@@ -82,14 +77,32 @@ func main() {
 	chkfatal(err)
 	srv := makeHandler(&config, NewDaemon(state))
 	http.Handle("/", srv)
-	if config.WebProtocol == "http" {
+
+	switch config.Insecure {
```

Comment:
Now that this is a boolean, it should probably go back to being an if statement. The default branch is dead code now too.